### PR TITLE
Rename InputStream to Pull

### LINKED ISSUE #1461: Find another name for ZStream.InputStream
Too easy to conflate with `java.io.InputStream`.

cc @jdegoes @vasilmkd @mschuwalow and all other naming experts :-)

### PR DESCRIPTION
**Pull request by vasilmkd**
Also, address stack safety by changing ZStream from being a trait to a class.

Resolves #1461.

@iravid @jdegoes

### CODE REVIEW

File: streams/shared/src/main/scala/zio/stream/ZStream.scala [hunk 1]
```diff
@@ -2138,18 +2100,16 @@ object ZStream extends ZStreamPlatformSpecific {
...
 
   /**
-   * Creates a stream from a scoped [[InputStream]].
+   * Creates a stream from a scoped [[Pull]].
    */
-  final def fromInputStreamManaged[R, E, A](is: ZManaged[R, E, InputStream[R, E, A]]): ZStream[R, E, A] =
-    new ZStream[R, E, A] {
-      def process = is
-    }
+  final def fromPullManaged[R, E, A](pull: ZManaged[R, E, Pull[R, E, A]]): ZStream[R, E, A] =
```
Comment (iravid) [inline streams/shared/src/main/scala/zio/stream/ZStream.scala]: Let's rename this to `def apply`

Comment (vasilmkd) [reply streams/shared/src/main/scala/zio/stream/ZStream.scala]: `def apply` and keep curly braces or go with the new `ZStream()`?

Comment (iravid) [reply streams/shared/src/main/scala/zio/stream/ZStream.scala]: The definition here has to be `def apply(pull: ..) = new ZStream(pull)`, but in the call sites you can then do:

```scala
ZStream {
  <expression that returns pull>
}
```

**Comment by vasilmkd** [comment]
@iravid Ready for review.

**Pull request merged**